### PR TITLE
feat(p2p): refactor downloaders and add peer id to the result

### PR DESCRIPTION
**Pull request by rkrasiuk**
Refactor downloaders, move them into a single package, and add peer id to download results. This will later be used in peer penalization. Closes #231

### CODE REVIEW

File: crates/interfaces/src/p2p/error.rs [hunk 1]
```diff
@@ -1,8 +1,12 @@
+use reth_primitives::WithPeerId;
 use tokio::sync::{mpsc, oneshot};
 
 /// Result alias for result of a request.
 pub type RequestResult<T> = Result<T, RequestError>;
 
+/// Result with [PeerId]
+pub type PeerRequestResult<T> = Result<WithPeerId<T>, RequestError>;
```
Comment (mattsse) [inline crates/interfaces/src/p2p/error.rs:8]:
```suggestion
pub type PeerRequestResult<T> = RequestResult<WithPeerId<T>>;
```

File: crates/primitives/src/peer.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    /// Get the peer id
+    pub fn peer_id(&self) -> PeerId {
+        self.0
+    }
+
+    /// Get the underlying data
+    pub fn data(&self) -> &T {
+        &self.1
+    }
+
```
Comment (mattsse) [inline crates/primitives/src/peer.rs:29]: would like a `split(self) -> (Id, T)` here

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nits

File: crates/primitives/src/peer.rs (hunk 2 above)
Comment (rkrasiuk) [reply crates/primitives/src/peer.rs:29]: done!

**Review by mattsse — APPROVED** [review]
lgtm

**Review by onbjerg — APPROVED** [review]

**Pull request merged**